### PR TITLE
Add version information in some functions

### PR DESCRIPTION
**Pull request by tomschr**
This PR contains some minor fixes in docstrings only:

* Use `.. versionadded::` RST directive in docstrings to make it more visible when something was added
* Minor wording fix in docstrings (versions -> version strings)

### CODE REVIEW

File: semver.py [hunk 1]
```diff
@@ -433,7 +442,7 @@ def _compare_by_keys(d1, d2):
 
 def compare(ver1, ver2):
     """
-    Compare two versions.
+    Compare two versions strings.
 
     :param ver1: version string 1
```
Comment (ppkt) [inline semver.py]: A quick question / suggestion - maybe in such case we should also add typing? Like for instance: 
```suggestion
    :param str ver1: version string 1
```
IDEs (like PyCharm) would love that feature :)

Comment (tomschr) [reply semver.py]: Generally you are right. :+1:
However, I'm not sure. We have already issue #213 which is targeted for major version 3. So maybe, I'd suggest to wait until we can release the next major release? Would that be ok?

Comment (ppkt) [reply semver.py]: Yeah, sure :)

**Comment by tomschr** [comment]
As this PR is only about docstrings, it shouldn't create any problems, so merging it.

**Pull request merged**